Lib/test/test_frame.py

```diff
@@ -364,6 +364,22 @@ def f():
             c = 0
         f()
 
+    def test_local_objects(self):
+        o = object()
+        k = '.'.join(['a', 'b', 'c'])
+        f_locals = sys._getframe().f_locals
+        f_locals['o'] = f_locals['k']
+        self.assertEqual(o, 'a.b.c')
+
+    def test_update_with_self(self):
+        # Make sure reference is not leaking here
```

Comment:
The problem was not a leak though -- it was a free-after-use (the opposite of a leak, really :-).

Reply:
Right. I was imaging a pool of reference and a hole that's leaking the reference without people knowing it so the pool is drained, but yeah leak is not a good word choice here. I just deleted it, the test itself makes sense without any comments.

Reply:
Ooh, that's actually a nice image. :-) I will merge now.

Reply:
Thanks for the quick review!